### PR TITLE
Dp 12413 add input percentage

### PR DESCRIPTION
**Pull request by clairesunstudio**

### CODE REVIEW

File: react/src/components/atoms/forms/InputNumber/index.js [hunk 1]
```diff
@@ -85,19 +86,24 @@ const NumberInput = (props) => (
           return(
             <div className="ma__input-number">
               <input {...inputAttr} />
+              {
+                <span className="ma__input-number-unit">{props.unit}</span>
```
Comment (clairesunstudio) [inline react/src/components/atoms/forms/InputNumber/index.js]:
```suggestion
                props.unit && <span className="ma__input-number-unit">{props.unit}</span>
```

File: react/src/components/atoms/forms/InputNumber/index.js [hunk 2]
```diff
@@ -85,19 +86,24 @@ const NumberInput = (props) => (
           return(
             <div className="ma__input-number">
               <input {...inputAttr} />
+              {
+                props.unit && <span className="ma__input-number-unit">{props.unit}</span>
```
Comment (avrilpearl) [inline react/src/components/atoms/forms/InputNumber/index.js]: i would add a condition to not render this if the value is null, i.e. when the placeholder is being rendered

File: react/src/components/utilities/componentPropTypeCheck.js [hunk 3]
```diff
@@ -80,3 +80,10 @@ export const validateFilters = (
 };
 
 export default componentPropTypeCheck;
+
+export const singleCharacterPropTypeCheck = (props, propName) => {
+  const prop = props[propName];
+  if (typeof prop !== 'string' || prop.length > 1) {
+    return new Error(`${propName} needs to be a single character string.`);
```
Comment (avrilpearl) [inline react/src/components/utilities/componentPropTypeCheck.js]: i would make this more flexible, at least 2 length, e.g. if someone wanted to use other units such as kg, cm, km, ft, yd

**Review by avrilpearl — CHANGES_REQUESTED** [review]
see comments above

**Review by avrilpearl — APPROVED** [review]
LGTM

**Pull request merged**